File: IOU/DTOs/ExpenseDTO.cs
namespace Api.IOU.DTOs;

public class ExpenseDTO
{
    public int Id { get; set; }
    public int SessionId { get; set; }
    public int PaidById { get; set; }
    public decimal TotalAmount { get; set; }
    public string Description { get; set; } = string.Empty;
    public List<ExpenseSplitDTO> Splits { get; set; } = new List<ExpenseSplitDTO>();
}

File: FrontEnd/iou-ui/src/app/Models/expenseDto.ts
import { ExpenseSplitDto } from "./expenseSplitDto";

export interface ExpenseDto {
  id: number;
  description: string;
  totalAmount: number;
  paidById: number;
  paidByUsername: string; // optional, helps display the name
  isEqualSplit: boolean;
  splits: ExpenseSplitDto[]; // each participant's owed amount
  createdAt: string; // optional
}